bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -1270,10 +1279,14 @@ void DWARFRewriter::updateDWARFObjectAddressRanges(
     }
 
     if (RangesBaseInfo) {
-      DIEBldr.replaceValue(&Die, RangesBaseInfo.getAttribute(),
-                           RangesBaseInfo.getForm(),
-                           DIEInteger(static_cast<uint32_t>(*RangesBase)));
-      RangesBase = std::nullopt;
+      if (RangesBaseInfo.getAttribute() == dwarf::DW_AT_GNU_ranges_base) {
+        UpdatedDIEsByDWO[*Unit.getDWOId()] = &Die;
```

Comment:
Not setting RangesBase = std::nullopt;
Can you check why this isn't necessary anymore. If it's a bug add a test to cover this.